code/game/gamemodes/events/space_weather.dm

```diff
@@ -0,0 +1,252 @@
+
+/datum/event/bluespace_storm/end()
+	command_announcement.Announce("The bluespace storm has ended.", "Bluespace Storm")
+
+/datum/event/bluespace_storm/tick()
+	if(prob(2))
+		var/area/A = random_ship_area(filter_maintenance = TRUE, filter_critical = TRUE)
+		bluespace_distorsion(A.random_space())
+
+/datum/storyevent/ion_blizzard
+	id = "ion blizzard"
+	name = "Ion blizzard"
+	weight = 0.9
+	event_type = /datum/event/ion_blizzard
+	parallel = FALSE
+	event_pools = list(EVENT_LEVEL_MUNDANE = POOL_THRESHOLD_MUNDANE, EVENT_LEVEL_MODERATE = POOL_THRESHOLD_MODERATE)
+	tags = list(TAG_SCARY, TAG_NEGATIVE)
+
+/datum/event/ion_blizzard
+	startWhen = 1
+	announceWhen = 1
+	endWhen = 350
+
+/datum/event/ion_blizzard/setup()
+	endWhen = rand(300, 600)
+
+/datum/event/ion_blizzard/announce()
+	command_announcement.Announce("A severe ion storm has been detected near the ship. Lighting subsystems are currently overloaded and may not work properly.", "Ion Blizzard")
+
+/datum/event/ion_blizzard/end()
+	command_announcement.Announce("The ion blizzard has ended.", "Ion Blizzard")
+
+/datum/event/ion_blizzard/tick() //get random ship area and do things to all light in there
+	if(prob(10)) //don't check every single light every single tick jesus christ
+		for(var/obj/machinery/light/L in random_ship_area())
+			L.broken()
+	else if(prob(80))
+		for(var/obj/machinery/light/L in random_ship_area())
+			L.flick_light(rand(2,5))
+
+/datum/storyevent/photon_vortex
+	id = "photon vortex"
+	name = "Photon vortex"
+	weight = 0.9
+	event_type = /datum/event/photon_vortex
+	parallel = FALSE
+	event_pools = list(EVENT_LEVEL_MUNDANE = POOL_THRESHOLD_MUNDANE, EVENT_LEVEL_MODERATE = POOL_THRESHOLD_MODERATE)
+	tags = list(TAG_SCARY, TAG_NEGATIVE)
+
+/datum/event/photon_vortex
+	startWhen = 2
+	announceWhen = 1
+	endWhen = 350
+
+/datum/event/photon_vortex/setup()
+	endWhen = rand(300, 600)
+
+/datum/event/photon_vortex/start()
+	for(var/obj/item/device/lighting/L in ship_areas)
+		L.brightness_on = L.brightness_on / 3
+		L.update_icon()
+	for(var/obj/machinery/light/l in ship_areas)
+		l.brightness_range = l.brightness_range / 3
+		l.brightness_power = l.brightness_power / 2
+		l.update()
+
+/datum/event/photon_vortex/announce()
+	command_announcement.Announce("A photon vortex anomaly has been detected near the ship. All photon-emitting machinery gives much less light.", "Photon Vortex Anomaly")
+
+/datum/event/photon_vortex/end()
+	command_announcement.Announce("The photon vortex anomaly has moved away from the ship.", "Photon Vortex Anomaly")
+
+	for(var/area/area as anything in ship_areas)
+		for(var/obj/item/device/lighting/L in area)
+			L.brightness_on = initial(L.brightness_on)
+			L.update_icon()
+		for(var/obj/machinery/light/l in ship_areas)
+			l.brightness_range = initial(l.brightness_range)
+			l.brightness_power = initial(l.brightness_power)
+			l.update()
```

Comment:
Flashlights are not in the area list. I see three options here:
1) Keep track of those items separately (meaning having a list, to which all non-lightbulb light sources are added on creation, and removed from on deletion).
2) Loop trough `world` as you did before.
3) Leave flashlights alone.

Pick which evil you think is lesser.